### PR TITLE
feat(client): show and execute backend project tests

### PR DESCRIPTION
**Pull request by ValeraS**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.

Related #17548

Need to write tests for [apis-and-microservices-projects](https://github.com/freeCodeCamp/freeCodeCamp/tree/master/curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects) and [information-security-and-quality-assurance-projects](https://github.com/freeCodeCamp/freeCodeCamp/tree/master/curriculum/challenges/english/06-information-security-and-quality-assurance/information-security-and-quality-assurance-projects)

### CODE REVIEW

**Comment by raisedadead** [comment]
While QA I found that the current state of `master` is completely broken for the backend projects.

This error seems to be pre-existing on master and I am not sure where it managed to creep in from (thanks no integration tests!).

image


@ValeraS can you please take a look what's going wrong? Its not linked to your PR, but you are the closest one to the issue.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
I tested the API Project: Timestamp Microservice project with the example site and a partially completed one of my own and they seem to be working nicely!  The tests pass and fail when expected to.  I'm not 100% confident I understand every detail, though, so I'll give it another look when I can. 

There is a problem with the front end projects, outlined in my comment.

File: client/src/templates/Challenges/project/ProjectForm.js [hunk 1]
```diff
@@ -1,32 +1,18 @@
...
-  // makeRequired,
-  // createFormValidator
-} from '../../../components/formHelpers';
+import { Form } from '../../../components/formHelpers';
 
 const propTypes = {
   isFrontEnd: PropTypes.bool,
   isSubmitting: PropTypes.bool,
-  openModal: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
```
[the remark this answers is not in the document]
Comment (ValeraS) [reply client/src/templates/Challenges/project/ProjectForm.js]: fixed

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by ojeytonwilliams** [comment]
Everything LGTM now!  One thing that would help during QA is just to push any changes as new commits, rather than rebasing.  Just to make changes easier to track.

The PR looks good, though.  Backend projects can be tested and frontend projects behave like they do in production.

**Comment by raisedadead** [comment]
> Need to write tests for [apis-and-microservices-projects](https://github.com/freeCodeCamp/freeCodeCamp/tree/master/curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects) [](/freeCodeCamp/freeCodeCamp/tree/HEAD@{2019-03-25T07:55:07Z}/curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects) and [information-security-and-quality-assurance-projects](https://github.com/freeCodeCamp/freeCodeCamp/tree/master/curriculum/challenges/english/06-information-security-and-quality-assurance/information-security-and-quality-assurance-projects) [](/freeCodeCamp/freeCodeCamp/tree/HEAD@{2019-03-25T07:55:07Z}/curriculum/challenges/english/06-information-security-and-quality-assurance/information-security-and-quality-assurance-projects)

Just to confirm, this PR still needs this before we can land this on master? If yes, would @freeCodeCamp/moderators be interested in adding those tests? This will be a great improvement to land.

**Comment by christophrus** [comment]
> > Need to write tests for [apis-and-microservices-projects](https://github.com/freeCodeCamp/freeCodeCamp/tree/master/curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects)  and [information-security-and-quality-assurance-projects](https://github.com/freeCodeCamp/freeCodeCamp/tree/master/curriculum/challenges/english/06-information-security-and-quality-assurance/information-security-and-quality-assurance-projects)
> 
> Just to confirm, this PR still needs adding the tests before we can land this on master? If yes, would @freeCodeCamp/moderators be interested in adding those tests? This PR will be a great improvement to land.

Tests are already written, see: https://fcc-back-end-tester.glitch.me

**Review by raisedadead — DISMISSED** [review]

**Comment by RandellDawson** [comment]
@christophrus It is my understanding that the Glitch app you referenced is not quite up to date and many of the tests are either not correct or some of the projects are not tested.

**Comment by ValeraS** [comment]
@raisedadead 

> Just to confirm, this PR still needs adding the tests before we can land this on master?

No, we can land this PR without tests and add tests in other PRs. Where we do not have tests, Campers pass projects without any checks as before.

**Comment by raisedadead** [comment]
> [...] I compiled this glitch app together from all the different broken or uncomplete glitch apps that I could find and fixed the obvious errors of them while I was testing my projects when I went through the curriculum like 2-3 month ago. So this app includes tests for all current projects and dont has any errors that I would know about.

I think @christophrus wanted to tag @RandellDawson instead. 

@christophrus This is some amazing work, we would love your help in bringing some of those tests and fixes to the main repo here and the glitch projects too if needed. I'll ask @scissorsneedfoodtoo and @RandellDawson to help you and guide with this further.

Essentially we would want to have tests for all the projects so that they are tested as per the functionality added in this PR.

Let us know your thoughts in this.

> No, we can land this PR without tests and add tests in other PRs. Where we do not have tests, Campers pass projects without any checks as before.

@ValeraS thanks for confirming, I'll pull this down and land this soon.

**Review by raisedadead — APPROVED** [review]

**Comment by christophrus** [comment]
@raisedadead Yeah I would try to do it, but I would definitly need some assistance with it. First of all I need to know how the current state is. Can I simply set up an fcc dev enviroment, put in the rules into the specific .md files and then try to debug them? Or do I have to turn some switch to get the rules from the .md files applied?

**Comment by raisedadead** [comment]
Sure, we are here to help you help us. 

You should hit us up in [Contributors Chat room](https://gitter.im/FreeCodeCamp/Contributors) for a more real time chat. @RandellDawson and @ojeytonwilliams and a few others including @scissorsneedfoodtoo (from the dev team) can guide you on the approach.

**Comment by RandellDawson** [comment]
@christophrus Are you still interested in adding the tests from the glitch project?  Just reach out to us on Gitter via the [Contributors Chat room](https://gitter.im/FreeCodeCamp/Contributors) or DM on Gitter.

**Comment by christophrus** [comment]
@RandellDawson Yeah absolutely, found it kind of hard to get an overview about what's going on in this gitter thingy and eventually got distracted by other things I moved on to.

**Comment by RandellDawson** [comment]
@christophrus If you have a forum account, you can also send a DM to me over there.  I have the same username on the [forum](https://www.freecodecamp.org/forum).

**Comment by scissorsneedfoodtoo** [comment]
@christophrus, your compilation of all of the broken Glitch projects sound great, and your tests for these projects will be really helpful to other learners.

**Comment by raisedadead** [comment]
Hi all. Just chiming in to say, let's discuss this over a new thread so that its tracked instead of this PR thread, for sake of context of the new action items.

**Comment by moT01** [comment]
Bump https://github.com/freeCodeCamp/freeCodeCamp/issues/36467

**Comment by raisedadead** [comment]
> let's discuss this over a new thread so that its tracked instead of this PR

**Pull request merged**